core/arch/arm/plat-imx/drivers/tzc380.c

```diff
@@ -11,17 +11,27 @@
 #include <imx.h>
 #include <mm/core_memprot.h>
 #include <mm/generic_ram_layout.h>
+#include <config.h>
+
+#define TZASC2_OFFSET		0x4000
```

Comment:
I don't really like this solution you came up with.
I am not a big fan of defining offset register in a .c file, there are specific header files for this : plat-imx/registers/*.h
Plus, this offset is defined for all boards and CPUs, even if they do not feature a TZASC2 controller.
TZASC2_BASE is already defined and I think it should be used instead.

IMO you are just moving the problem : it's like directly defining TZASC2_BASE in this file.

What do you think?

Reply:
@clementfaure This is easily fixed by moving this into the header file. `TZASC2_BASE` could also be used, at the expense of adding a define for other platforms which sets this to `0`. You can even add an `assert(TZASC2_BASE != 0)` to ensure that no unsupported board runs into this at runtime. Unless OP-TEE adds its own device tree entries and probes them from the device tree, this is IMO the proper way to support the special case where two TZC380 IP cores have been implemented on the SoC.

Reply:
I moved the `TZASC2_BASE` to `imx-regs.h` and the assert to not hit this code path unintentionally.